### PR TITLE
213555/Additional-detail-to-messages

### PR DESCRIPTION
**Pull request by Kirsten1822**
Ticket link: https://dev.azure.com/defragovuk/DEFRA-FFC/_sprints/taskboard/DEFRA-FFC%20Payments%20Team/DEFRA-FFC/Payment%20Service/Sprint%2011?workitem=213555

### CODE REVIEW

**Comment by Kirsten1822** [comment]
Acceptance criteria:

- return files for IMPS, Glos and Genesis can be processed by Responses
- all responses are published as messages to return topic
- messages are validated
- a property of `ledger` should be added to all responses with AP as the value
- sample return files from Transformation Layer can be processed

File: app/processing/parse-genesis-return-file.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const moment = require('moment')
+const { convertToPence } = require('../currency-convert')
+
+const parseGenesisReturnFile = (csv, filename) => {
+  return csv.map(x => {
+    const row = x.split('^')
+    if (row[0] === 'D') {
+      return {
+        // Record type, payment id, transaction number, value, date, payment type, payment reference, status, reason
```
Comment (johnwatson484) [inline app/processing/parse-genesis-return-file.js:9]: Probably can delete each of these comments now?

File: app/processing/parse-imps-return-file.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      return {
+        // Record type, payment job number, fes code, trader number, transaction number, status, payment reference, gbp value, payment type, date, eur value, exchange rate
+        sourceSystem: 'IMPS',
+        paymentJobNumber: row[1],
+        fesCode: row[2],
+        traderNumber: row[3],
+        transactionNumber: row[4],
+        settled: row[5] === 'P',
+        reference: row[6],
+        valueGBP: convertToPence(row[7]),
```
Comment (johnwatson484) [inline app/processing/parse-imps-return-file.js:17]: Update to just `value` instead of `valueGBP`

File: app/processing/parse-return-file.js [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
+    return parseGenesisReturnFile(csv, filename)
+  }
+  if (filename.includes('FCAP')) {
+    return parseGlosReturnFile(csv, filename)
+  }
+  if (filename.includes('RET_IMPS')) {
+    return parseImpsReturnFile(csv, filename)
+  } else {
+    return parseDefaultReturnFile(csv, filename)
+  }
```
Comment (johnwatson484) [inline app/processing/parse-return-file.js:18]: Don't really need the else, can just be `return parseDefaultReturnFile(csv, filename)` as the other ifs all return anyway if they're hit.

File: app/processing/parse-genesis-return-file.js (hunk 1 above)
Comment (Kirsten1822) [reply app/processing/parse-genesis-return-file.js:9]: Yep I can do

**Review by johnwatson484 — APPROVED** [review]

**Pull request merged**